specs/agents/metrics.md

```diff
@@ -50,11 +50,109 @@ Agents should not send the `system.process.cgroup.memory.mem.limit.bytes` metric
+
+Agents SHOULD attempt to assign the appropriate label value based on the counted event. If this would impose significant implementation overhead, the value `other` MUST be used instead.
+
+All event count metrics are enabled/disabled via the `agent_event_metrics` configuration:
+
+|                |   |
+|----------------|---|
+| Type           | `boolean` |
+| Default        | `true` |
+| Dynamic        | `true` |
+| Central config | `true` |
+
+#### Event Queue Metrics
+
+The following metrics SHOULD be recorded by agents regarding the state of the internal event reporting queue:
+
+ - `agent.events.queue.capacity`: The number of events the reporting queue can hold at maximum
+ - `agent.events.queue.min_size`: The smallest number of events the queue contained since this metric was reported the last time
+ - `agent.events.queue.max_size`: The biggest number of events the queue contained since this metric was reported the last time
+
+We capture both `min_size` and `max_size` to allow to distinguish between reasons for why the queue is becoming full:
+ * There is too much data captured / network errors: Both `min_size` and `max_size` are at `capacity`
+ * The data is collected in bursts for which the queue is not correctly sized: `max_size` is at `capacity`, but `min_size` is low.
+
+All event queue metrics are enabled/disabled via the `agent_event_metrics` specified [above](#event-count-metrics).
+
+#### Event Request Metrics
+
+Agents SHOULD expose the following metrics regarding Intake API networking:
+
+ - `agent.events.requests.count`: `COUNTER` of the number of requests made or attempted to the Intake API of the APM server
+ - `agent.events.requests.bytes`: `COUNTER` of the approximate number of bytes on the wire sent to the Intake API of the APM
+
+All these metrics MUST have the label `success`, which can have the values `true` or `false`. A request is counted with `success=true` iff the APM Server responded with `2xx`.
+
+The metric `agent.events.requests.bytes` does not need to represent the exact network usage.
+Instead the number of compressed bytes within the request body can be used as approximation.
+
+All event request metrics are enabled/disabled via the `agent_event_metrics` specified [above](#event-count-metrics).
+
+#### Agent Overhead Metrics
+
+If possible with a reasonable amount of runtime and implementation overhead, agents SHOULD expose the following metrics for approximating their own resource usage:
+
+ - `agent.cpu.overhead.pct`: The fraction of the process cpu usage caused by the agent in a range [0-1] within the last reporting interval.
+ - `agent.cpu.total.pct`: The approximate CPU usage caused by the agent. Derived by multiplying `agent.cpu.overhead.pct` with `system.process.cpu.total.norm.pct`.
+ - `agent.memory.allocation.bytes`: `COUNTER` of the number of bytes allocated in the main memory by the agent.
```

Comment:
Maybe use a prefix like `agent.background.*` to make it clearer that this doesn't attempt to capture the full overhead of the agent but just its background activity. The descriptions and labels in the dashboard should also reflect that.